src/platform/macos/mod.rs

```diff
@@ -871,7 +871,7 @@ impl OsIpcOneShotServer {
 }
 
 pub struct OsIpcSharedMemory {
-    ptr: *mut u8,
+    ptr: Option<NonNull<u8>>,
     length: usize,
```

Comment:
Does shared memory of size 0 make any sense? How about using [NonZeroUsize](https://doc.rust-lang.org/nightly/std/num/type.NonZeroUsize.html) for the length and just `NonNull<u8>` for the ptr?

Reply:
I was also wondering about that, but then we do not allow empty shared memory. One nicer option would be to simply disallow non-zero `OsIpcSharedMemory` but allow it in `IpcSharedMemory`, so we have:
```rust
pub struct IpcSharedMemory {
    os_shared_memory: Option<OsIpcSharedMemory>,
}
```